### PR TITLE
gate: dedupe the blocking-severity scan + revert-failing guard for the no-fanout label carve-out (#1643)

### LINKED ISSUE #1643: gate: dedupe roundCarriesBlockingSeverity scan + guard requireFanoutEvidence:false non-application (deferred #1621 follow-ups)
## Summary

Two low-severity findings were deferred (per the repo's `blockCleanOnFindingSeverities: [high]` policy) during the #1621 gate drive (PR #1642, merged). They are non-blocking but worth tracking durably (the gate findings ledgers live under ephemeral `tmp/`). Both are still present on main.

## Deferred findings

1. **`roundCarriesBlockingSeverity` duplicates the structured-findings blocking-severity scan** (`scripts/github/upsert-checkpoint-verdict.mjs`). The per-angle `for(angle)/for(f)` loop, the `RESOLVED_DISPOSITIONS` skip, the unparseable-entry tally, and the `normalizeSeverity` + blocking-severity check are duplicated between `roundCarriesBlockingSeverity` (line 1548) and the clean-verdict refusal's structured-findings cross-check (lines 1933-1971). The two can drift if the disposition-skip or severity-normalization rule changes in one but not the other. Refactor: extract one shared scan helper used by both.

   Contract notes for the extraction. The two call sites consume different shapes: the escalation path needs a boolean, the clean-verdict path needs the list of blocking severities observed plus an `unparseableBlocking` flag for its refusal message. A helper such as `scanBlockingFindings(structuredFindings, blocking)` returning `{ blockingObserved, unparseableBlocking }` covers both. Two behaviors must survive intact: the `RESOLVED_DISPOSITIONS` skip applies only to parseable findings, never to `angle.unparseable` entries (an unparseable finding carries no disposition to resolve against); and the helper covers only the structured-findings scan. The `findingsSeverityCounts` and `verdict === "findings_present"` fallbacks (lines 1571-1574) stay in `roundCarriesBlockingSeverity`.

2. **No revert-failing test guards the `requireFanoutEvidence:false` non-application of the gate:full label.** The `escalateGateFullLabel` guard (line 2156) short-circuits on `resolveRequireFanoutEvidence(config)`, but no test asserts `gateFullLabelApplied === undefined` for a `findings_present` inline round under `requireFanoutEvidence:false` (grep for `gateFullLabelApplied` in `test/` finds nothing). Removing that guard would apply the label to repos that opted out of fan-out and no test would fail. Extend the existing #1621 derive test (`test/github/upsert-checkpoint-verdict.test.mjs:6250`, "a findings_present draft_gate verdict DERIVES the next action"), which already stages `requireFanoutEvidence:false` via `stageConfigRepoRoot(false)` and captures gh calls via `makeAcRunChild`, to also assert `result.gateFullLabelApplied === undefined` and that `calls` contains no `label create` and no `pr edit ... --add-label` entry.

## Acceptance criteria

- [x] One shared helper in `scripts/github/upsert-checkpoint-verdict.mjs` performs the per-angle blocking-severity scan; `roundCarriesBlockingSeverity` and the clean-verdict cross-check both call it, and neither retains its own `for(angle)/for(f)` loop over `structuredFindings`.
- [x] The helper skips parseable findings whose disposition is in `RESOLVED_DISPOSITIONS` and never applies the skip to `angle.unparseable` entries; both scans keep this asymmetry.
- [x] The clean-verdict refusal message is byte-identical to today's, including the UNPARSEABLE sentence; the existing #1526 test asserting `/UNPARSEABLE/i` still passes unchanged.
- [x] The `findingsSeverityCounts` and bare `findings_present` fallbacks remain in `roundCarriesBlockingSeverity` and behave as before when `structuredFindings` is absent.
- [x] A test asserts that a `findings_present` inline round with a blocking severity under `requireFanoutEvidence:false` yields `result.gateFullLabelApplied === undefined`.
- [x] The same test asserts no `gh label create` call and no `gh pr edit ... --add-label` call fired.
- [x] Reverting the `resolveRequireFanoutEvidence(config)` conjunct in `escalateGateFullLabel` makes the new test fail (verified once locally; the revert is not committed).

## Definition of done

- [x] `node --test test/github/upsert-checkpoint-verdict.test.mjs` green.
- [x] `npm run verify` green.
- [x] `npm run assets:check` and `npm run schema:check` green.
- [x] Refactor is behavior-preserving: no CLI output, exit-code, or posted-comment change; no new files or exports beyond the one helper.
- [x] Scope-bounded: only `scripts/github/upsert-checkpoint-verdict.mjs` and `test/github/upsert-checkpoint-verdict.test.mjs` change.

## Non-goals

- Changing the `blockCleanOnFindingSeverities: [high]` policy (medium/low deferral is intentional).
- The separate (also deferred) design question of whether a ready-PR all-fetch-failure should fail closed on the unticked-AC check (currently skips; the gate cannot verify what it cannot read; the draft-gate refinement check owns the "no spec-of-record" case).
- Changing gate:full label semantics or the escalation trigger conditions themselves (#1621 landed those; this issue only adds the missing revert-failing guard).
- Touching the mirrored copy under `ombase/`.

## Refined (2026-08-17)

Auto-grill against main 81e86581: both findings verified still present (duplication at `upsert-checkpoint-verdict.mjs:1548` vs 1933-1971; zero `gateFullLabelApplied` references in tests); line pins and test anchors updated from current source.

### PR DESCRIPTION
**Pull request by mfittko**
## Summary

Two deferred low findings from the verdict-precondition drive, landed together: the blocking-severity scan over structured findings now lives in one shared helper used by both `roundCarriesBlockingSeverity` and the clean-verdict cross-check (they could previously drift), and a revert-failing test now guards the `requireFanoutEvidence:false` non-application of the gate:full label.

## Scope and context

Two files. `scripts/github/upsert-checkpoint-verdict.mjs` gains the module-local `scanBlockingFindings(structuredFindings, blocking)` returning the observed blocking severities plus an unparseable-blocking flag; both former inline loops now call it. The disposition-skip asymmetry survives (resolved dispositions skip parseable findings only, never unparseable entries), the severity-counts and bare findings_present fallbacks stay in `roundCarriesBlockingSeverity`, and the clean-verdict refusal message is byte-identical. Two test edits: the existing derive test gains assertions that a findings_present inline round under `requireFanoutEvidence:false` applies no gate:full label (result field undefined; no label-create or add-label gh call), and the parseable-only clean-refusal test gains a negative pin asserting its message is UNPARSEABLE-free (binding the helper's unparseable-blocking flag in both directions).

## Acceptance criteria

- [x] One shared helper performs the per-angle blocking-severity scan; both call sites use it and neither retains its own loop.
- [x] The helper skips resolved parseable findings only; unparseable entries are never skipped.
- [x] The clean-verdict refusal message is byte-identical, UNPARSEABLE sentence included; the existing regex test passes unchanged.
- [x] The severity-counts and bare findings_present fallbacks remain in roundCarriesBlockingSeverity.
- [x] A test asserts gateFullLabelApplied is undefined for a findings_present inline round under requireFanoutEvidence:false.
- [x] The same test asserts no label-create and no add-label gh call fired.
- [x] Reverting the guard conjunct makes the new test fail (verified locally, revert not committed: strict-equal failure observed, plus six pre-existing tests also failing on the reverted guard).

## Definition of done

- [x] `node --test test/github/upsert-checkpoint-verdict.test.mjs` green (168 tests).
- [x] `npm run verify` green (gate validation artifact at head).
- [x] `npm run assets:check` (exit 0) and `npm run schema:check` (exit 0).
- [x] Behavior-preserving: no CLI output, exit-code, or posted-comment change; one module-local helper, no new exports.
- [x] Scope-bounded to the two files.

## AC / DoD / Non-goals matrix

| Acceptance criterion | Verified by DoD item(s) | Non-goal boundary |
|---|---|---|
| Shared scan helper, both call sites, no duplicate loops | upsert-checkpoint-verdict suite green (168) | blockCleanOnFindingSeverities policy unchanged |
| Disposition-skip asymmetry preserved | suite green; refusal message byte-identical | gate:full semantics unchanged |
| Severity-counts and bare findings_present fallbacks unchanged | suite green (fallback tests untouched) | ombase mirror untouched |
| Revert-failing label guard test | revert verified locally (not committed) | all-fetch-failure fail-closed question stays open |
| Full verification | npm run verify, assets:check, schema:check | ombase mirror untouched |

## Non-goals

- Changing the `blockCleanOnFindingSeverities` policy (medium/low deferral is intentional).
- The separate deferred question whether a ready-PR all-fetch-failure should fail closed on the unticked-AC check.
- Changing gate:full label semantics or escalation triggers.
- Touching the mirrored copy under ombase/.

## Open questions

None.

## Validation

```sh
node --test test/github/upsert-checkpoint-verdict.test.mjs
npm run verify
```

Closes #1643

### CODE REVIEW

File: scripts/github/upsert-checkpoint-verdict.mjs [hunk 1]
```diff
@@ -1538,6 +1538,34 @@ function deriveEffectiveNextAction(verdict, gate) {
...
+    }
+    for (const u of angle.unparseable ?? []) {
+      const sev = /** @type {string} */ (normalizeSeverity(u.severity));
+      if (blocking.includes(sev)) {
+        observed.add(sev);
+        unparseableBlocking = true;
+      }
+    }
+  }
+  return { blockingObserved: blocking.filter((sev) => observed.has(sev)), unparseableBlocking };
```
Comment (mfittko) [inline scripts/github/upsert-checkpoint-verdict.mjs:1566]: **low** (`correctness`): Not fully behavior-preserving for the clean-verdict cross-check: the old tally only counted a severity when it was a key of observedCounts, i.e. a member of the canonical SEVERITY_ORDER vocabulary, so a configured blocking severity outside that vocabulary could never land in blockingObserved and a clean verdict passed. The helper uses an unrestricted Set, so such a severity now refuses the clean verdict. Reachable because the config loader returns the merged raw config when zod validation fails and the poster discards the returned errors, so an out-of-enum blockCleanOnFindingSeverities value survives resolveGateConfig unfiltered. The direction is fail-closed and it matches the untouched escalation path, which always used a bare blocking.includes, so this removes a real drift rather than creating one, but it contradicts the definition-of-done claim that the refactor changes no behavior.

File: scripts/github/upsert-checkpoint-verdict.mjs [hunk 2]
```diff
@@ -1538,6 +1538,34 @@ function deriveEffectiveNextAction(verdict, gate) {
...
+    for (const f of angle.findings) {
+      if (RESOLVED_DISPOSITIONS.has(f.disposition)) continue;
+      const sev = /** @type {string} */ (normalizeSeverity(f.severity));
+      if (blocking.includes(sev)) observed.add(sev);
+    }
+    for (const u of angle.unparseable ?? []) {
+      const sev = /** @type {string} */ (normalizeSeverity(u.severity));
+      if (blocking.includes(sev)) {
+        observed.add(sev);
+        unparseableBlocking = true;
```
Comment (mfittko) [inline scripts/github/upsert-checkpoint-verdict.mjs:1562]: **low** (`coverage`): The unparseableBlocking flag has presence coverage but no absence coverage. The clean-verdict refusal appends the UNPARSEABLE sentence only when the blocking hit came from an angle.unparseable entry, and one test asserts that sentence is present for an unparseable blocking finding. No test asserts it is ABSENT when the refusal is caused by parseable findings only: the three parseable-only refusal tests assert only the Cannot set verdict clean text, the unresolved findings text, and the high severity token, never a doesNotMatch on UNPARSEABLE. Moving the flag into the shared helper puts the parseable loop and the unparseable loop adjacent, so hoisting the assignment one loop up would mis-attribute every parseable refusal as UNPARSEABLE and the whole 168-test suite would still pass. This is message-only mis-attribution, no verdict change, so severity is low.

**Review by mfittko — COMMENTED** [review]
### Gate review: `draft_gate`


**Reviewed head SHA:** `071279ce1c1792e466816ee1e9b1f26c727a34cd`
**Verdict:** clean
**Execution mode:** fanout_fanin

**Findings summary:** 8 angles reviewed; 3 findings (see per-angle breakdown below).

- `contract-surface` → `clean`
- `correctness` → `findings_present` (1 finding)
- `coverage` → `findings_present` (1 finding)
- `determinism` → `clean`
- `gate-evidence` → `clean`
- `input-validation` → `clean`
- `pr-description` → `findings_present` (1 finding)
- `scope` → `clean`

**Body-filed findings** (no in-diff location):


> **nit** (`pr-description`): The PR body omits the repo template's dedicated Linked issue heading; the closing keyword for issue 1643 sits as a trailing line after the Validation code fence instead of under its own section. Purely cosmetic template compliance: the closing keyword still resolves, and the gate context already resolved acceptanceCriteria to issue 1643 with source linked-issue.

**Next action:** Apply the one judge act (negative UNPARSEABLE assertion), re-gate at the fixed head, then mark ready

**Review by mfittko — COMMENTED** [review]
### Gate review: `draft_gate`


**Reviewed head SHA:** `37dd265481abb5df6db5fe9b41ee150569f8cc55`
**Verdict:** clean
**Execution mode:** fanout_fanin

**Findings summary:** 8 angles reviewed; no findings (see per-angle breakdown below).

- `correctness` → `clean`
- `coverage` → `clean`
- `determinism` → `clean`
- `gate-evidence` → `clean`
- `pr-description` → `clean`
- `contract-surface` → `clean`
- `scope` → `clean`
- `input-validation` → `clean`

**Body-filed findings** (no in-diff location):

> None — every finding this round is an inline comment on this review.

**Next action:** Mark the PR ready for review (round 2 clean; 3 angles carried from the prior clean round)

File: scripts/github/upsert-checkpoint-verdict.mjs (hunk 1 above)
Comment (mfittko) [reply scripts/github/upsert-checkpoint-verdict.mjs:1566]: Deferred at gate close (round 2, fingerprint 41a3a98f514e1a37, severity low, angle correctness): low findings are deferred at gate close after the fixer triaged them (fix-if-cheap-in-the-same-commit, else defer); the deferral is recorded on this thread's marker and in the durable findings ledger.

File: scripts/github/upsert-checkpoint-verdict.mjs (hunk 2 above)
Comment (mfittko) [reply scripts/github/upsert-checkpoint-verdict.mjs:1562]: Deferred at gate close (round 2, fingerprint 6c76890ae65d1656, severity low, angle coverage): low findings are deferred at gate close after the fixer triaged them (fix-if-cheap-in-the-same-commit, else defer); the deferral is recorded on this thread's marker and in the durable findings ledger.

**Review by mfittko — COMMENTED** [review]
### Gate review: `pre_approval_gate`


**Reviewed head SHA:** `37dd265481abb5df6db5fe9b41ee150569f8cc55`
**Verdict:** clean
**Execution mode:** fanout_fanin

**Findings summary:** 5 angles reviewed; no findings (see per-angle breakdown below).

- `acceptance-criteria` → `clean`
- `contradiction-lens` → `clean`
- `pr-checklist-matrix` → `clean`
- `renderer-security` → `clean`
- `yagni` → `clean`

**Body-filed findings** (no in-diff location):

> None — every finding this round is an inline comment on this review.

**Next action:** Merge under the recorded standing authorization (ADR 0050): full gate pass complete
